packages/dashboard/src/components/SubwayNav/SubwayNav.tsx

```diff
@@ -0,0 +1,298 @@
+      return <React.Fragment />;
+    }
+  }
+
+  /**
+   * Render step of the subway Nav control
+   * @param stepToRender
+   * @param prevStep
+   */
+  private _renderStep(stepToRender: ISubwayNavStep, prevStep: ISubwayNavStep | undefined): JSX.Element[] {
+    let stepElements: JSX.Element[] = [];
+
+    if (stepToRender === undefined) {
+      return stepElements;
+    }
+
+    const { steps } = this.props;
+
+    if (steps !== undefined) {
+      const iconProps = this._getIconPropsStep(stepToRender);
+
+      if (prevStep !== undefined) {
+        const connectorClass = this._getStepConnectorClassName(stepToRender);
+        stepElements.push(
+          <div className={css(classNames.subwayNavStepConnector, connectorClass)} key={'connector' + '-' + stepToRender.key} />
+        );
+      }
+
+      if (stepToRender !== undefined) {
+        const renderSubStep = stepToRender.state === SubwayNavStepState.Current && this._hasSubSteps(stepToRender);
+
+        stepElements.push(
+          <div
+            key={stepToRender.key}
+            className={css(
+              classNames.subwayNavStepDiv,
+              stepToRender.state !== SubwayNavStepState.Current && stepToRender.disabled ? classNames.disableStep : undefined
+            )}
+          >
+            <Icon iconName={iconProps.iconName} className={css(iconProps.iconClassName, classNames.subwayNavStepIcon)} />
+            <Link onClick={() => stepToRender.onClickStep(stepToRender, undefined)}>
+              <span
+                className={css(classNames.stepLabel, stepToRender.state === SubwayNavStepState.Current ? classNames.boldStep : undefined)}
+              >
+                {stepToRender!.label}
+              </span>
+            </Link>
+          </div>
+        );
+
+        /** Render substeps if current step has substeps */
+        if (renderSubStep && stepToRender!.subSteps) {
+          let subStepElements: JSX.Element[] = [];
+
+          stepElements.push(
+            <div
+              className={css(classNames.subwayNavStepConnector, classNames.stepConnectorCompleted)}
+              key={'connector-sub' + '-' + stepToRender.key}
+            />
+          );
+
+          let subSteps = stepToRender.subSteps;
+          let prevStep: ISubwayNavStep | undefined = undefined;
+
+          if (subSteps !== undefined) {
+            subSteps.forEach((subStep: ISubwayNavStep, index: number) => {
+              subStepElements = subStepElements.concat(this._renderSubStep(stepToRender, subStep, index > 0 ? prevStep : undefined));
+              prevStep = subStep;
+            });
+
+            stepElements = stepElements.concat(subStepElements);
+          }
+        }
+      } // if stepToRender
+    } // if steps !== undefined
+
+    return stepElements;
+  }
+
+  /**
+   * Render substeps of current step in subway Nav control
+   * @param parentStep
+   * @param subStepToRender
+   * @param prevSubStep
+   */
+  private _renderSubStep(
```

Comment:
I wonder if a substep can be treated like a decorated step so that you can merge these two rendering into one.

They seems pretty similar.

Reply:
In my code it's the same component with different styling applied. It's handled as a state instead of a different component.